syft/get_source_config.go

```diff
@@ -0,0 +1,115 @@
+package syft
+
+import (
+	"crypto"
+	"fmt"
+
+	"github.com/anchore/go-collections"
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft/source"
+)
+
+type GetSourceConfig struct {
+	// SourceProviderConfig may optionally be provided to be used when constructing the default set of source providers, unused if SourceProviders specified
+	SourceProviderConfig SourceProviderConfig
+
+	// SourceProviders the tagged set of known source.Provider to use, will use Syft's default set if not provided
+	SourceProviders []collections.TaggedValue[source.Provider]
+
+	// BaseSources is used to restrict the set of sources prior to user-provided
+	BaseSources []string
+
+	// FromSource is an explicit list of source names to use, in order, to attempt to locate a source
+	FromSource []string
+
+	// DefaultImageSource will cause a particular image pull source to be used as the first pull source, followed by other pull sources
+	DefaultImageSource string
+}
+
+func (c GetSourceConfig) WithAlias(alias source.Alias) GetSourceConfig {
+	c.SourceProviderConfig.Alias = alias
+	return c
+}
+
+func (c GetSourceConfig) WithRegistryOptions(registryOptions *image.RegistryOptions) GetSourceConfig {
+	c.SourceProviderConfig.RegistryOptions = registryOptions
+	return c
+}
+
+func (c GetSourceConfig) WithPlatform(platform *image.Platform) GetSourceConfig {
+	c.SourceProviderConfig.Platform = platform
+	return c
+}
+
+func (c GetSourceConfig) WithExcludeConfig(excludeConfig source.ExcludeConfig) GetSourceConfig {
+	c.SourceProviderConfig.Exclude = excludeConfig
+	return c
+}
+
+func (c GetSourceConfig) WithDigestAlgorithms(algorithms ...crypto.Hash) GetSourceConfig {
+	c.SourceProviderConfig.DigestAlgorithms = algorithms
+	return c
+}
+
+func (c GetSourceConfig) WithBasePath(basePath string) GetSourceConfig {
+	c.SourceProviderConfig.BasePath = basePath
+	return c
+}
```

Comment:
these all wrap options that already exist on the source provider config, so I feel that those should in place of providing these:
```golang 
cfg.WithSourceProviderConfig(DefaultSourceProvider().WithAlias("thing"))
```

Reply:
I'm pretty sure that the main reason we'd like to include the `.With***` types of methods is to make specifying the configuration more convenient and discoverable. The most likely usage is to call `syft.GetSource`, so the configuration for that call should be convenient and discoverable, which would be less so if the user had to deal with only the inner config in most cases. I _would_ use struct embedding here, but need the return type to be `GetSourceConfig`, not `SourceProviderConfig`.